apps/freeablo/faworld/inventory.cpp

```diff
@@ -19,6 +19,190 @@ using namespace boost::container;
 
 namespace FAWorld
 {
+    Inv::Inv(int32_t width, int32_t height, bool treatAllIItemsAs1by1) : mInventoryBox(width, height), mTreatAllItemsAs1by1(treatAllIItemsAs1by1)
+    {
+        for (uint8_t y = 0; y < mInventoryBox.height(); y++)
+        {
+            for (uint8_t x = 0; x < mInventoryBox.width(); x++)
+            {
+                mInventoryBox.get(x, y).mInvX = x;
+                mInventoryBox.get(x, y).mInvY = y;
+            }
+        }
+    }
+
+    bool Inv::autoPlaceItem(Item& item, PlacementCheckOrder order)
```

Comment:
Well here's definitely going to be a conflict. I had to separate autoplacement function so it would return location first and I could check that item fits before buying and then place it to appropriate location.